api-design-guidelines/index.md

```diff
@@ -93,7 +93,9 @@ is printed.
       great summary.
 
     * **Use a single sentence fragment** if possible, ending with a
-      period.  Do not use a complete sentence.
+      period.  Do not use a complete sentence.  A summary is like 
+      a dictionary definition; it is shown in the context of the 
+      API it describes.
```

Comment:
on Fri Feb 19 2016, Brent Royal-Gordon <notifications-AT-github.com> wrote:

on Fri Feb 19 2016, Brent Royal-Gordon <notifications-AT-github.com> wrote:

>> @@ -93,7 +93,9 @@ is printed.
>>        great summary.
>>  
>>      * **Use a single sentence fragment** if possible, ending with a
>> -      period.  Do not use a complete sentence.
>> +      period.  Do not use a complete sentence.  A summary is like 
>> +      a dictionary definition; it is shown in the context of the 
>> +      API it describes.
>
>> Isn't this obvious?  I mean, the summary is in the doc comment, which is
> right there next to the declaration.
>
> It should be, and yet sometimes when I'm writing a doc comment I have
> to remind myself of this. (I think this mainly happens when I'm
> documenting a type—I have an impulse to start "The ImageSource
> struct…")

I don't see how this change helps reduce the impulse; it sounds just
like justification for overriding the impulse when you do have it.

> Maybe it can just be worded as "Don't restate the API's name—it'll be
> obvious from the context."

Don't the examples already make clear what form these summaries should
take?

I'm not trying to be difficult here, but the guidelines should epitomize
“omit needless words” as well as “clarity matters.”  The guidelines
intentionally omit rationales; if we put rationales in everywhere it
would be a much longer document, with lots of subjective assertions
(e.g. about the value of fluent APIs), which I think would weaken it.

I understand wanting the rationales to be available, though; maybe
there's a way to present them without weakening the whole document... a
little box labeled “Why?” that you can hover over and see an
explanation?  If there are any web designers out there who want to
prototype something, we could discuss it.

---
Reply to this email directly or view it on GitHub:
https://github.com/apple/swift-internals/pull/5/files#r53511780